### PR TITLE
Switch portable scripting projects to project.json

### PR DESCRIPTION
**Pull request by tmat**
@jasonmalinowski @agocke 

once IC v37 is in we can push this.

### CODE REVIEW

File: build/Targets/VSL.Settings.targets [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
     <RoslynSemanticVersion Condition="'$(RoslynSemanticVersion)' == ''">1.1.0</RoslynSemanticVersion>
     <SystemReflectionMetadataAssemblyVersion Condition="'$(SystemReflectionMetadataAssemblyVersion)' == ''">1.1.0</SystemReflectionMetadataAssemblyVersion>
-    <SystemCollectionsImmutableAssemblyVersion Condition="'$(SystemCollectionsImmutableAssemblyVersion)' == ''">1.1.36</SystemCollectionsImmutableAssemblyVersion>
+    <SystemCollectionsImmutableAssemblyVersion Condition="'$(SystemCollectionsImmutableAssemblyVersion)' == ''">1.1.37</SystemCollectionsImmutableAssemblyVersion>
```
Comment (jasonmalinowski) [inline build/Targets/VSL.Settings.targets]: FYI it's probably going to be one more day before we land support for this in the main VS source tree.

File: src/Scripting/CSharp/CSharpScripting.csproj [hunk 2]
```diff
@@ -12,13 +11,9 @@
     <RootNamespace>Microsoft.CodeAnalysis.Scripting.CSharp</RootNamespace>
     <AssemblyName>Microsoft.CodeAnalysis.Scripting.CSharp</AssemblyName>
     <SolutionDir Condition="'$(SolutionDir)' == '' OR '$(SolutionDir)' == '*Undefined*'">..\..\..\</SolutionDir>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>
+    <TargetFrameworkProfile></TargetFrameworkProfile>
+    <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
     <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
-    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
```
Comment (jasonmalinowski) [inline src/Scripting/CSharp/CSharpScripting.csproj]: I suspect you still want the flavoring, correct @davkean and @rchande?

Comment (rchande) [inline src/Scripting/CSharp/CSharpScripting.csproj]: Yes, leave these.

Comment (tmat) [inline src/Scripting/CSharp/CSharpScripting.csproj]: OK. Will (re)add them.

**Pull request closed without merging**